### ISSUE TITLE
[Bug] httpx doesn't support int header values

### ISSUE THREAD

**Issue by asermax** [fern-api/fern#2698]
**Describe the bug**
I have a Header value defined as an `int` in a FastAPI endpoint:
```python
async def register_block_completed_action(user_id: int = Header(Required, alias="X-UserId")):
  ...
```
FastAPI translates as a header param of type `integer` in its OpenAPI output:
```json
    ...
    "parameters": [
        {
            "required": true,
            "schema": {
                "title": "X-Userid",
                "type": "integer"
            },
            "name": "X-UserId",
            "in": "header"
        }
    ],
    ...
```
Then Fern generates the client with the correct types:
```python
    async def register_block_completed_action(
        self, *, user_id: int
    ) -> typing.List[UserActionResponse]:
        _response = await self._client_wrapper.httpx_client.request(
            "POST",
            urllib.parse.urljoin(f"{self._client_wrapper.get_base_url()}/", "some/url"),
            headers=remove_none_from_dict({**self._client_wrapper.get_headers(), "X-UserId": user_id}),
            timeout=60,
        )
        ...
```

The problem lies when trying to use the client, as apparently `httpx` expects headers to be either `str` or `bytes`, which breaks when we pass in the `user_id` as an `int` (this is in the context of a test suite I ran, I don't include the traceback for brevity):
```python
    def normalize_header_value(
        value: typing.Union[str, bytes], encoding: typing.Optional[str] = None
    ) -> bytes:
        """
        Coerce str/bytes into a strictly byte-wise HTTP header value.
        """
        if isinstance(value, bytes):
            return value
>       return value.encode(encoding or "ascii")
E       AttributeError: 'int' object has no attribute 'encode'
```

**Expected behavior**
I think the best solution would be to cast any headers to `str`; that way we get to keep the correct typing for the API client while making sure `httpx` gets what it expects:
```python
    async def register_block_completed_action(
        self, *, user_id: int
    ) -> typing.List[UserActionResponse]:
        _response = await self._client_wrapper.httpx_client.request(
            "POST",
            urllib.parse.urljoin(f"{self._client_wrapper.get_base_url()}/", "some/url"),
            headers=remove_none_from_dict({**self._client_wrapper.get_headers(), "X-UserId": str(user_id)}),
            timeout=60,
        )
        ...
```

**Versions**
 - Fern CLI: 0.16.19
 - Generator: 1.0.0-rc0
 
 **Additional context**
This doesn't seem too hard to fix, so I might make a PR myself, but wanted to leave the issue in case I don't get to it in the short-term :sweat_smile:

**Comment by dannysheridan** [comment]
@asermax it'd be great to see a PR from you on this!

**Comment by armandobelardo** [comment]
@asermax is this with the python-sdk generator? Or the FastAPI generator?

If the python-sdk generator, this should now be fixed within the latest generator ([1.1.0-rc4](https://github.com/fern-api/fern/blob/main/generators/python/sdk/CHANGELOG.md#110-rc3---2024-04-04))
